### ISSUE TITLE
Webserver crashes after few API calls from external server after upgrade to 0.15.8

### ISSUE THREAD

**Issue by aballiet** [kestra-io/kestra#3402]
### Describe the issue
Webserver was working fine in 0.14.4, after upgrading to 0.15.8, webserver systematically crashes after few API calls (2 calls).

We tried a bunch of things and noticed the server struggles globally.

Here is an example of a flow which created a lot of scaling events on Kestra server and can make it crash :
```YAML
id: kestra-webserver-crasher
namespace: dev
tasks:
- id: spike-webserver-memory
  type: io.kestra.plugin.scripts.python.Script
  runner: PROCESS
  beforeCommands:
  - "pip install requests > /dev/null"
  warningOnStdErr: false
  docker:
    pullPolicy: ALWAYS
  script: |
    import requests

    BASE_SERVICE_URL = "http://kestra-service.kestra:8080/"

    # Get all running excections
    search = {"size":20}
    response = requests.get(f"{BASE_SERVICE_URL}/api/v1/executions/search", params=search)
    total = response.json()["total"]
    if total == 0:
        print("No executions currently running")
    else:
        print(f"{total} executions to fetch logs for")
        executions = [
            {"id": result["id"], "flow": result["flowId"], "namespace": result["namespace"]}
            for result in response.json()["results"]
        ]
        for execution in executions:
            id = execution["id"]
            log_search = {"q": ""}
            logs = requests.get(f"{BASE_SERVICE_URL}/api/v1/logs/{id}", params=log_search)
            print(logs)
```

### Environment

- Kestra Version: 0.15.8 (also tested with 0.15.9, same issue)
- Operating System (OS/Docker/Kubernetes): Kubernetes
- Java Version (if you don't run kestra in Docker):

**Comment by loicmathieu** [comment]
Hi,
Do you have some logs on the Kestra webserver side?

**Comment by aballiet** [comment]
updated the issue @loicmathieu

**Comment by loicmathieu** [comment]
There is a workaround with the following configuration:

```yaml
configuration:
  micronaut:
    server:
      max-request-size: 1GB
      netty:
        server-type: full_content
```

But it would limit the request size to 1GB and silently ignore bigger requests, this can be an issue if you start a flow with total inputs size of more than 1GB.

We're still working with the Micronaut team to find a proper fix, see https://github.com/micronaut-projects/micronaut-core/issues/10677

**Comment by tchiotludo** [comment]
@aballiet do you still have the issues?

**Comment by yuri1969** [comment]
Just leaving a note here - v0.16.10 is also affected.

**Comment by loicmathieu** [comment]
@aballiet @yuri1969 just discovered a bug in Micronaut (not easily reproducible) with POST request including a payload when returning a 404 not found that hang leading to memory / CPU issue. I wonder if it could be the cause.

So, do you se any webhook that you call with a POST method and a payload? Do you sometimes, by inadvertance, use it with a flow or webhook key non-existent?

**Comment by yuri1969** [comment]
@loicmathieu As far as our deployment is concerned it starts executions exclusively by POSTing to `/api/v1/executions/<namespace>/<flow_id>` - no webhooks are used. Nearly every flow defines two required inputs of the FILE type.

Occasional misconfiguration incidents lead to POSTing to a nonexistent `<namespace>/<flow_id>` flow which might trigger HTTP 404 responses.

**Comment by loicmathieu** [comment]
I stumble upon a bug when an exception is returned when creating the execution, see https://github.com/kestra-io/kestra/pull/4836

**Comment by loicmathieu** [comment]
We fix a bunch of issues with file upload if 0.19, can you try 0.19.10 and see if it fixes the issue?

**Comment by yuri1969** [comment]
@loicmathieu Sadly, our deployed version is still way behind v0.19.

**Comment by Ben8t** [comment]
@aballiet did you get a chance to try on newest versions ?

**Comment by loicmathieu** [comment]
For those who have the issue and wanted to help, they can turn on paranoid netty leak detector and send any logs it would emit.

```
netty:
  resource-leak-detector-level: paranoid
```

**Comment by Ben8t** [comment]
@loicmathieu just for my knowledge, why did you reopen the issue?

**Comment by loicmathieu** [comment]
This issue is a different memory leak due to FILE input upload, it is not linked to the fix I did for the thread leak on the follow endpoint.

**Comment by anna-geller** [comment]
I don't understand why this issue is still open, the fix is provided in the more recent versions, no feedback from users about it for 1/2 year and no action is needed here on our end (if I'm mistaken, what are the next steps on our end?)

> This issue is a different memory leak due to FILE input upload, it is not linked to the fix I did for the thread leak on the follow endpoint.
if it's a different, still unresolved memory leak, let's open a fresh new issue with a reproducer, logs, and your kestra version.

**Comment by loicmathieu** [comment]
No, the issue is still existing, we have a possible fix but we still need to validate that it worked.
